cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -2649,9 +2656,15 @@ private ApplicationMeta decodeRow(StructuredRow row) {
     ApplicationMeta meta = GSON.fromJson(
         row.getString(StoreDefinition.AppMetadataStore.APPLICATION_DATA_FIELD),
         ApplicationMeta.class);
-    SourceControlMeta sourceControl = GSON.fromJson(
-        row.getString(StoreDefinition.AppMetadataStore.SOURCE_CONTROL_META),
-        SourceControlMeta.class);
+    SourceControlMeta sourceControl = null;
+    try {
+      sourceControl = getSourceControlMetadataStore().getAppSourceControlMeta(
```

Comment:
Same as above question. New design segregates App spec and scm metadata. I am assuming scm metadata updates are needed only when there is a push or pull. Do we still need this logic?

Reply:
Here, we are not updating the scm metadata. We are simply retrieving the info.

Also scm metadata update is also needed for deploy operation which I have explained in the above comment. Please let me know if there are any queries.